docs/Extensions.md

```diff
@@ -16,19 +16,65 @@ The naming convention serves two purposes. Firstly it makes it clear for everyon
 ## General Structure
 Roc expects all extensions to export an object named `roc` from the main file in a npm package. Other than this the packages themselves are quite free to do whatever they want to create what is needed to build an application.
 
-The object must contain a name (most likely the same as the `name` defined in `package.json`) and at least one of the following:
+The object must contain a name (most likely the same as the `name` defined in `package.json`), optionally but highly recommended a `version`, and at least one of the following:
 ```
+packages        Array of packages that the extension inherits from
+plugins         Array of plugins that the extension inherits from
+
+init            A function will initialize the extension
+postInit        A function that will run after all extensions have been initialized in the reverse order as they was added
+
 buildConfig     Function that takes the previous configuration and meta objects and returns new ones
---- OR ---
 config          Configuration object
 meta            Meta configuration object
-
 actions         An object with actions that should connect to the hooks
 hooks           An object with the hooks that a package uses, not needed but useful for documentation and convenience
-packages        Array of packages that the extension inherits from
-plugins         Array of plugins that the extension inherits from
 ```
 __If `buildConfig` is specified, config and meta will be ignored.__
+__If `init` is specified, buildConfig, config, meta, actions and hooks will be ignored.__
+
+
+#### packages
+An array of Roc packages that the extension uses. Classically this will look like this:
+```
+packages: [
+    require.resolve('roc-package-core-dev'),
+    require.resolve('roc-package-module')
+]
+```
+
+#### plugins
+An array of plugins that the extension uses. Classically this will look like this:
+```
+plugins: [
+    require.resolve('roc-plugin-start'),
+]
+```
+
+#### init
+Can be used to initialize the extension. Will get the following as an object:
+```
+config          The configuration object at this time
+meta            The meta configuration object at this time
+extensions      The registered extensions at this time, is an array with the following structure: [{name, version}]
+actions         The registered actions at this time
+hooks           The registered hooks at this time
+```
+If the extension should be processed the function need to return an object that can have one of the following from above; `buildConfig`, `config`, `meta`, `actions` and `hooks`. If the function returns nothing/undefined it will be managed as an error and the extension and all that depends on it will not be loaded. This is also the case if it returns a string, the string can be used for a more detailed error message.
```

Comment:
Not 100% sure of best way to be explicit here, but we could for example say
"one of the following from general structure section above" <- with a clickable link to that section.